src/system/SystemLayer.h

```diff
@@ -327,7 +328,12 @@ class EventLoopHandler : public chip::IntrusiveListNodeBase<>
     intptr_t mState = 0;
 };
 
-class LayerSocketsLoop : public LayerSockets
+class LayerSocketsLoop :
+#if CHIP_SYSTEM_CONFIG_USE_SOCKETS
```

Comment:
so we have a class `LayerSocketsLoop` but that seems to have conditions based on `CHIP_SYSTEM_CONFIG_USE_SOCKETS` ... that seems odd since `Sockets` is right in the name.

What is going on here and should we find a better name then?

Reply:
Good point. I reverted the change from this PR. And I will refine it in following PRs. This PR is now only focusing on refactoring the wake event.